### ISSUE TITLE
macos metal: #include "leptonica/allheaders.h"

### ISSUE THREAD

**Issue by dickreuter** [sirfz/tesserocr#316]
pip install tesserocr currently doesn't work (and neither does conda). I get an error:
```
      tesserocr.cpp:796:10: fatal error: 'leptonica/allheaders.h' file not found
      #include "leptonica/allheaders.h"
```

**Comment by zdenop** [comment]
Did you install leptonica?

**Comment by dickreuter** [comment]
Yes, Leptonica is installed, as described here: https://macappstore.org/leptonica/

**Comment by zdenop** [comment]
Can you please modify `#include "leptonica/allheaders.h"` to `#include "allheaders.h"`

**Comment by dickreuter** [comment]
where do I need to make this change?

**Comment by zdenop** [comment]
`tesserocr.cpp` line 796 - it is in error message

**Comment by dickreuter** [comment]
Where do i find tesserocrp.cpp? It's not in the github repo and I don't think it's on my machine before I pip install tesserocr.

**Comment by zdenop** [comment]
tesserocr.cpp is generated file by setup.py ;-)

So can you please provide a full log of the following commands?

```
git clone https://github.com/sirfz/tesserocr.git
cd tesserocr
python3 setup.py build
```

**Comment by dickreuter** [comment]
(base) dickr@NICOLASs-MacBook-Pro tesserocr % python setup.py build
Supporting tesseract v5.3.0
Tesseract major version 5
Building with configs: {'libraries': ['tesseract', 'lept'], 'compile_time_env': {'TESSERACT_MAJOR_VERSION': 5, 'TESSERACT_VERSION': 84082688}}
running build
running build_ext
Compiling tesserocr.pyx because it changed.
[1/1] Cythonizing tesserocr.pyx
Detected compiler: unix
building 'tesserocr' extension
creating build
creating build/temp.macosx-11.1-arm64-cpython-39
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/dickr/opt/anaconda3/include -arch arm64 -I/Users/dickr/opt/anaconda3/include -fPIC -O2 -isystem /Users/dickr/opt/anaconda3/include -arch arm64 -I/Users/dickr/opt/anaconda3/include/python3.9 -c tesserocr.cpp -o build/temp.macosx-11.1-arm64-cpython-39/tesserocr.o -std=c++11 -DUSE_STD_NAMESPACE
tesserocr.cpp:796:10: fatal error: 'leptonica/allheaders.h' file not found
#include "leptonica/allheaders.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
error: command '/usr/bin/clang' failed with exit code 1

**Comment by dickreuter** [comment]
Any suggestions what we can do about this?

**Comment by zdenop** [comment]
First of all: I am not mac user ;-)

Can you run and provide output of  `pkg-config --libs --cflags lept` ?

Generally you need to specify path where compiler (clang) can find leptonica header file. So first of all please find there is located "allheaders.h" file.

Then you will need something like this (I found it on internet, so I hope it is correct):
```
export CPLUS_INCLUDE_PATH="${CPLUS_INCLUDE_PATH:+${CPLUS_INCLUDE_PATH}:}<path_to_leptonica>"
```

+ you can try this:
```
brew install tesseract
export LIBRARY_PATH="/opt/homebrew/lib"
export CPATH="/opt/homebrew/include"
```

After this setting try to install tesserocr (`python setup.py build`).

**Comment by dickreuter** [comment]
```
pkg-config --libs --cflags lept
-I/opt/homebrew/Cellar/leptonica/1.82.0_1/include/leptonica -L/opt/homebrew/Cellar/leptonica/1.82.0_1/lib -llept
```

I tried 
`CPPFLAGS=-I/opt/homebrew/Cellar/leptonica/1.82.0_1/include/leptonica  pip install tesserocr`
But same error.

I also tried:

```
export LIBRARY_PATH="CPPFLAGS=-I/opt/homebrew/Cellar/leptonica/1.82.0_1/include"
export CPATH ="CPPFLAGS=-I/opt/homebrew/Cellar/leptonica/1.82.0_1/lib"
```

but same error


Any additional suggestions are greatly appreciated.

**Comment by dickreuter** [comment]
export LIBRARY_PATH="CPPFLAGS=-I/opt/homebrew/Cellar/leptonica/1.82.0_1/include/leptonica"

worked. thanks

**Comment by isaackatayev** [comment]
For me, doing the 2nd part of @zdenop 's solution worked
`
export LIBRARY_PATH="/opt/homebrew/lib"
export CPATH="/opt/homebrew/include"
`
